pyomo/contrib/fbbt/fbbt.py

```diff
@@ -1130,21 +1153,22 @@ def visiting_potential_leaf(self, node):
                    and may give the wrong solution. Even if the correct solution is found, this could 
                    introduce numerical problems.
                 """
-                lb = max(math.floor(lb), math.ceil(lb - self.integer_tol))
-                ub = min(math.ceil(ub), math.floor(ub + self.integer_tol))
-                if lb < value(node.lb):
-                    lb = value(node.lb)  # don't make the bounds worse than the original bounds
-                if ub > value(node.ub):
-                    ub = value(node.ub)  # don't make the bounds worse than the original bounds
+                if lb > -interval.inf:
+                    lb = max(math.floor(lb), math.ceil(lb - self.integer_tol))
+                if ub < interval.inf:
+                    ub = min(math.ceil(ub), math.floor(ub + self.integer_tol))
+                orig_lb = value(node.lb)
+                orig_ub = value(node.ub)
+                if orig_lb is None:
+                    orig_lb = -interval.inf
+                if orig_ub is None:
+                    orig_ub = interval.inf
+                if lb < orig_lb:
+                    lb = orig_lb  # don't make the bounds worse than the original bounds
+                if ub > orig_ub:
+                    ub = orig_ub  # don't make the bounds worse than the original bounds
```

Comment:
A comment similar to the above would be helpful.  Also, given that this is repeated code, would a helper function be warranted?